openeogeotrellis/backend.py

```diff
@@ -1899,46 +1901,58 @@ def get_results(self, job_id: str, user_id: str) -> Dict[str, dict]:
             ml_model_metadata['asset'] = False
             results_dict['ml_model_metadata.json'] = ml_model_metadata
 
-        if os.path.isfile(job_dir / 'out'):
-            results_dict['out'] = {
-                # TODO: give meaningful filename and extension
-                "asset": True,
-                "output_dir": str(job_dir),
-                "type": media_type,
-                "bands": bands,
-                "nodata": nodata
-            }
-
-        if os.path.isfile(job_dir / 'profile_dumps.tar.gz'):
-            results_dict['profile_dumps.tar.gz'] = {
-                "asset": True,
-                "output_dir": str(job_dir),
-                "type": "application/gzip"
-            }
-
-        for file_name in os.listdir(job_dir):
-            if file_name.endswith("_metadata.json") and file_name != JOB_METADATA_FILENAME:
-                results_dict[file_name] = {
+        # Updating the asset info based on what is on the file system was a solution for some old edge cases.
+        # For a kuberenetes deployment we can skip this, the edge cases should not happen here.
+        # On Kubernetes the result files with be put in the object storage instead of the container,
+        # because the k8s pods can be replaced at any moment and then the files would be gone.
+        if not ConfigParams().is_kube_deploy:
```

Comment:
IMO `is_kube_deploy` is an anti-pattern (see #283), can we avoid introducing it here?

Reply:
I have removed the use of is_kube_deploy in the new code, in favor of a more specific setting: use_object_storage.

For now it is still uses the value of is_kube_deploy in reality, but at least it can now be split up, and I was able to add test coverage to test downloading with use_object_storage set to True/on and False/off.
It is just a question of deciding how exactly we want to configure use_object_storage. (For example, use a new environment variable, or set it True only if SWIFT_BUCKET has a value, etc.)